Dockerfile

```diff
@@ -1,136 +1,178 @@
+
+# Install docker-compose
+ARG DOCKER_COMPOSE_VERSION=1.24.0
+RUN curl -sLo /dependencies/docker-compose                                                                                 \
+    "https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-$(uname -s)-$(uname -m)" \
+    && chmod +x /dependencies/docker-compose
+
+# Install goss
+ARG GOSS_VERSION=0.3.7
+RUN curl -sLo /dependencies/goss                                                             \
+    "https://github.com/aelsabbahy/goss/releases/download/v${GOSS_VERSION}/goss-linux-amd64" \
+  && chmod +x /dependencies/goss
+
+# Install conftest
+ARG CONFTEST_VERSION=0.4.2
+RUN set -euxo pipefail; curl -sL "https://github.com/instrumenta/conftest/releases/download/v${CONFTEST_VERSION}/conftest_${CONFTEST_VERSION}_Linux_x86_64.tar.gz" \
+  | tar -xz                                                                                                                                                        \
+  && mv conftest /dependencies/
+
+# Install jq
+ARG JQ_VERSION=1.5
+RUN curl -sLo /dependencies/jq                                                     \
+    "https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64" \
+  && chmod +x /dependencies/jq
+
+# Install yq
+ARG YQ_VERSION=2.1.1
+RUN curl -sLo /dependencies/yq                                                       \
+    "https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64" \
+  && chmod +x /dependencies/yq
+
+# Install kustomize
+ARG KUSTOMIZE_VERSION=2.0.2
+RUN curl -sLo /dependencies/kustomize                                                                   \
+    "https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZE_VERSION}/kustomize_${KUSTOMIZE_VERSION}_linux_amd64" \
+  && chmod +x /dependencies/kustomize
+
+#--------------------------#
+# docker-gcloud-sdk        #
+#--------------------------#
+FROM debian:buster-slim AS docker-gcloud-sdk
 
 ENV CLOUD_SDK_VERSION 189.0.0
 
-RUN \
-  DEBIAN_FRONTEND=noninteractive \
-    apt update && apt install --assume-yes --no-install-recommends \
-      apt-transport-https \
-      awscli \
-      bzip2 \
-      ca-certificates \
-      curl \
-      dnsutils \
-      gawk \
-      gettext-base \
-      git \
-      gnupg \
-      golang \
-      lsb-release \
-      lsof \
-      make \
-      nmap \
-      nmap-common \
-      ncat \
-      openssh-client \
-      parallel \
-      postgresql-client \
-      rsync \
-      wget \
-      xmlstarlet \
-  \
-  && export CLOUD_SDK_REPO \
-  && CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" \
+RUN apt-get update                                                               \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+      apt-transport-https                                                        \
+      awscli                                                                     \
```

Comment:
I like the lining up of the newline escape because its much easier to read when you're viewing locally in vim.. the change to it is hard to read in the diff but thenceforth it makes the diffs super clean.  See simulator ... https://github.com/kubernetes-simulator/simulator/blob/master/Dockerfile

Reply:
The more important change here was the alphabetising of the installed dependencies

Reply:
https://www.vim.org/scripts/script.php?script_id=294